test/tests/amqp/test_amqp_racknode.py

```diff
@@ -0,0 +1,205 @@
+            mon_url = '/api/2.0/nodes/{}'.format(rackid)
+            mon_data = fit_common.rackhdapi(mon_url)
+            self.assertIn(
+                mon_data['status'], range(200, 205),
+                "Incorrect HTTP return code: {}".format(mon_data['status']))
+            json_node_data = mon_data['json']
+            self.assertTrue(
+                json_node_data['name'] == newrack['name'] and json_node_data['type'] == "rack",
+                "rack node field error")
+            timecount = 0
+            while amqp_message_received is False and timecount < 10:
+                sleep(1)
+                timecount = timecount + 1
+            self.assertNotEquals(timecount, 10, "No AMQP message received")
+            expected_key = "node.added.information." + rackid + '.' + rackid
+            expected_payload = {
+                "type": "node",
+                "action": "added",
+                "typeId": rackid,
+                "nodeId": rackid,
+                "severity": "information",
+                "version": "1.0",
+                "createdAt": mon_data_post['json']['createdAt']}
+            self.assertEquals(self._compare_message(expected_key, expected_payload), True, "AMQP Message Check Error!")
+            logs.debug_3("query rack: %r successfully!" % rackname)
+        logs.debug_2("test: rack creation and query succeed!")
+
+    def test_api_delete_rack(self):
+        rack_node_list = test_api_utils.get_node_list_by_type("rack")
+        for rack in rack_node_list:
+            global amqp_message_received
+            amqp_message_received = False
+            # start amqp thread
+            td = AmqpWorker(
+                exchange_name="on.events", topic_routing_key="node.removed.#",
+                external_callback=self.amqp_callback, timeout=10)
+            td.setDaemon(True)
+            td.start()
+            mon_url = '/api/2.0/nodes/{}'.format(rack)
+            mon_data = fit_common.rackhdapi(mon_url, action='delete')
+            self.assertIn(
+                mon_data['status'], range(200, 205),
+                "Incorrect HTTP return code: {}".format(mon_data['status']))
+            timecount = 0
+            while amqp_message_received is False and timecount < 10:
+                sleep(1)
+                timecount = timecount + 1
+            self.assertNotEquals(timecount, 10, "No AMQP message received")
+            expected_key = "node.removed.information." + rack + '.' + rack
+            expected_payload = {
+                "type": "node",
+                "action": "removed",
+                "typeId": rack,
+                "nodeId": rack,
+                "severity": "information",
+                "version": "1.0",
+                "createdAt": mon_data['headers']['Date']}
+            self.assertEquals(self._compare_message(expected_key, expected_payload), True, "AMQP Message Check Error!")
+
+
+if __name__ == '__main__':
+    fit_common.unittest.main()

```

Comment:
We should not be using the unittest imported from fit_common.  This is something we are trying to fix along the way since a bad pattern was set earlier.  Please change this to unittest.main() and import unittest above.